### ISSUE TITLE
Improving the section on how to contribute

### ISSUE THREAD

**Issue by rcap107** [skrub-data/skrub#1130]
### Describe the issue linked to the documentation

Hello, while I was working on my first PR I hit a few snags and problems that may also happen to new contributors, so I thought it might be useful to update the section in the documentation relative to contributing to skrub. 

Some of the things I would like to add:
- [ ] A checklist of the operations that should be performed for the PR (pre-commit, write tests, update the changelog etc.)
- [ ] How to run tests using a specific environment with pixi on the local machine for debugging
- [ ] How to check for code coverage on the local machine 
- [ ] How to check doctests
- [ ] Preparing and running pre-commit
- [ ] Some information on where the code for testing is, and how to structure tests

If you have more ideas, please add them! 

@Vincent-Maladiere suggested having a section on working with pixi, but I don't have enough experience with it to write it.

**Comment by jeromedockes** [comment]
thanks @rcap107  that's a great idea. I guess this is the file to update: https://github.com/skrub-data/skrub/blob/main/CONTRIBUTING.rst

**Comment by jeromedockes** [comment]
@rcap107  could you update this issue to state what remains to be done after #1131? thanks

**Comment by MarieSacksick** [comment]
I considered tackling this issue before the sprint, but it looks like it's all done and that it can be closed :)!

**Comment by rcap107** [comment]
Yep, I'm finding some more things I'd like to change in the doc page, but everything I mentioned in the original post has been fixed so we can close this one.
